activestorage/test/service/gcs_service_test.rb

```diff
@@ -29,6 +29,30 @@ class ActiveStorage::Service::GCSServiceTest < ActiveSupport::TestCase
       @service.delete key
     end
 
+    test "direct upload with content disposition" do
+      key      = SecureRandom.base58(24)
+      data     = "Something else entirely!"
+      checksum = Digest::MD5.base64digest(data)
+      url      = @service.url_for_direct_upload(key, expires_in: 5.minutes, content_type: "text/plain", content_length: data.size, checksum: checksum)
+
+      uri = URI.parse url
+      request = Net::HTTP::Put.new uri.request_uri
+      request.body = data
+      @service.headers_for_direct_upload(key, checksum: checksum, filename: ActiveStorage::Filename.new("test.txt"), disposition: :attachment).each do |k, v|
+        request.add_field k, v
+      end
+      request.add_field "Content-Type", ""
```

Comment:
Is there a reason why we aren't including the checksum header here?

Reply:
It's not need to since `@service.headers_for_direct_upload(...)` above will include the checksum header (see [this line](https://github.com/rails/rails/pull/36793/files/788b51803924880253aa4b7ddbc1f3c4df3d2788#diff-bda6a610ef1575b2c8458c96b7f12578R110)).